pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -279,3 +284,40 @@ func (p *criStatsProvider) makeContainerStats(
 
 	return result
 }
+
+// removeTerminatedContainer returns the specified container but with
+// the stats of the terminated containers removed.
+func removeTerminatedContainer(containers []*runtimeapi.Container, stats []*runtimeapi.ContainerStats) []*runtimeapi.Container {
+	containerMap := make(map[containerID][]*runtimeapi.Container)
+	// Sort order by create time
+	sort.Slice(containers, func(i, j int) bool {
+		return containers[i].CreatedAt < containers[j].CreatedAt
+	})
+	for _, container := range containers {
+		refID := containerID{
+			podRef:        buildPodRef(container.Labels),
+			containerName: kubetypes.GetContainerName(container.Labels),
+		}
+		containerMap[refID] = append(containerMap[refID], container)
+	}
+
+	// Create container stats map.
+	containerStatsMap := make(map[string]*runtimeapi.ContainerStats)
+	for _, stat := range stats {
+		containerStatsMap[stat.Attributes.Id] = stat
+	}
+
+	result := make([]*runtimeapi.Container, 0)
+	for _, refs := range containerMap {
+		if len(refs) == 1 {
+			result = append(result, refs[0])
+			continue
+		}
+		for i := 0; i < len(refs); i++ {
```

Comment:
Can we add one more logic? If no running container found, we should add the latest container.

Reply:
Sorry for this, have added, PTAL, :-)